src/postprocessing.py

```diff
@@ -200,4 +258,123 @@ def crop_image_center_per_class(image, h_crop, w_crop):
+        iou_matrix = get_iou_matrix(category_instances, category_annotations)
+        for label_nr in range(1, category_instances.max() + 1):
+            mask = category_instances == label_nr
+            iou = get_iou(iou_matrix, label_nr)
+            mask_probabilities = np.where(mask, category_probabilities, 0)
+            area = np.count_nonzero(mask)
+            mean_prob = mask_probabilities.sum() / area
+            max_prob = mask_probabilities.max()
+            bbox = get_bbox(mask)
+            bbox_height = bbox[1] - bbox[0]
+            bbox_width = bbox[3] - bbox[2]
+            bbox_aspect_ratio = bbox_height / bbox_width
+            bbox_area = bbox_width * bbox_height
+            bbox_fill = area / bbox_area
+            dist_to_border = get_distance_to_border(bbox, mask.shape)
+            contour_length = get_contour_length(mask)
+            mask_features = {'iou': iou, 'threshold': threshold, 'area': area, 'mean_prob': mean_prob,
+                             'max_prob': max_prob, 'bbox_ar': bbox_aspect_ratio,
+                             'bbox_area': bbox_area, 'bbox_fill': bbox_fill, 'dist_to_border': dist_to_border,
+                             'contour_length': contour_length}
+            layer_features.append(mask_features)
+        image_features.append(pd.DataFrame(layer_features))
+    return image_features
+
+
+def get_iou_matrix(labels, annotations):
+    mask_anns = []
+    if annotations is None or annotations == []:
+        return None
+    else:
+        for annotation in annotations:
+            if not isinstance(annotation['segmentation'], dict):
+                annotation['segmentation'] = \
+                    cocomask.frPyObjects(annotation['segmentation'], labels.shape[0], labels.shape[1])[0]
+        annotations = [annotation['segmentation'] for annotation in annotations]
+        for label_nr in range(1, labels.max() + 1):
+            mask = labels == label_nr
+            mask_anns.append(rle_from_binary(mask.astype('uint8')))
+        iou_matrix = cocomask.iou(mask_anns, annotations, [0, ] * len(annotations))
+        return iou_matrix
+
+
+def get_iou(iou_matrix, label_nr):
+    if iou_matrix is not None:
+        return iou_matrix[label_nr - 1].max()
+    else:
+        return None
+
+
+def get_thresholds():
+    thresholds = []
+    for n_thresholds in CATEGORY_LAYERS:
+        thrs_step = 1. / (n_thresholds + 1)
+        category_thresholds = np.arange(thrs_step, 1, thrs_step)
+        thresholds.extend(category_thresholds)
+    return thresholds
+
+
+def get_bbox(mask):
+    '''taken from https://stackoverflow.com/questions/31400769/bounding-box-of-numpy-array and
+    modified to prevent bbox of zero area'''
+    rows = np.any(mask, axis=1)
+    cols = np.any(mask, axis=0)
+    rmin, rmax = np.where(rows)[0][[0, -1]]
+    cmin, cmax = np.where(cols)[0][[0, -1]]
+    return rmin, rmax + 1, cmin, cmax + 1
+
+
+def get_distance_to_border(bbox, im_size):
```

Comment:
@apyskir this is min distance to border right? maybe we could have one more feature with max distance to border as well

Reply:
@jakubczakon sure we can, I can add it